### PR TITLE
Don't override registries.conf by default

### PR DESCRIPTION
**Pull request by rhatdan**
We want to share registries.conf configuration between all of the tools using
containers/image and then allow crio.conf to override if the user wants, not
to override by default.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -48,7 +48,7 @@ github.com/mattn/go-runewidth v0.0.1
 github.com/seccomp/libseccomp-golang v0.9.0
 github.com/syndtr/gocapability e7cb7fa329f456b3855136a2642b197bad7366ba
 github.com/blang/semver v3.5.0
-github.com/BurntSushi/toml v0.2.0
+github.com/BurntSushi/toml b26d9c308763d68093482582cea63d69be07a0f0
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply vendor.conf]: Ok, I just grabbed what containers/image used.  Lets wait til the first round of testing finishes.

[the remark this answers is not in the document]
Comment (rhatdan) [reply vendor.conf]: The vndr changed nothing, but the vendor.conf will say v0.3.0 now.

**Comment by wking** [comment]
Build failures look [like][1]:

```
level=fatal msg="Near line 161 (last key parsed 'crio.image.docker.io'): expected key separator '=', but got '\n' instead"
```

Maybe a template-parsing issue?

[1]: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_critest_fedora/264/s3/download/journals/crio.service

**Comment by rhatdan** [comment]
No I know what it is, I will deal with it tomorrow.  You have to do some fancy sed commands to get the crio.conf in test environment correct.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Yes I got to talk to @mrunalp and @runcom about these. I can't see in the code where this is happening, but some tool is injecting "docker.io" into crio.conf around the spot where the registries are defined.  The problem is this patch comments out the registries definition in crio.conf by default so we end up with an uncommented docker.io in the crio.conf which is causing crio system service to crash.

We have a sed script that cleans this up in the helper.bash script for bats tests, but these seem to be somewhere else

**Comment by rhatdan** [comment]
@mrunalp @runcom If you look at the crio.conf file used in the tests
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_critest_fedora/266/s3/download/gathered/crio.conf
You will see the commented out # registries = 
line and then the docker.io that the tests added.

Maybe @cevich might have some insight on this as well.

**Comment by cevich** [comment]
Aye, grep saves the day.  I went through all the job definitions in OriginCI and didn't find any shenanigans happening from there.  It's gotta be the playbook doing it.

Note: I'm going through all the tasks, and fixing any that change files to also drop a comment that the change came from Ansible, and which YAML file.  I'll send up a separate PR for that.

**Comment by cevich** [comment]
For this one, are there any cases where that registry is actually needed?

One option (though very ugly) could be to add tags for that specific task, where the list of tags does *NOT* include ``e2e``.  Ugly because that means keeping tags in sync if they change/added.

**Comment by cevich** [comment]
Yeah, just checked, there's no way (other than on the command line) to specify a "negative" tag.

Another (cleaner) solution (if other tests need this registry config) is, we ADD a tag to that task, then also add it to the ansible-playbook command-line only for the jobs that need it (i.e. exclude the e2e job).

**Comment by cevich** [comment]
e.g.

```
- name: add docker.io default registry
  lineinfile:
    dest: /etc/crio/crio.conf
    line: |
        # Added by Ansible from build/cri-o.yml
        docker.io
    insertafter: 'registries = \['
    regexp: 'docker\.io'
    state: present
  tags:
    - config_docker_io
```

Then I can update all the non-e2e jobs in origin to also use that ``config_docker_io`` tag (and update the README.md here)

**Comment by rhatdan** [comment]
So the problem right now is we end up with
# registries = [
docker.io
# existing registries
# ]

We could just change the code to say.
    insertafter: 'registries = \['
    regexp: 'registries = [ "docker\.io" ]

Which should fix the problem.

**Comment by cevich** [comment]
OIC, it's a syntax snafu.  Really the best tool for this is to use ``template`` or ``blockinfile`` instead of ``lineinfile``.

File: contrib/test/integration/build/cri-o.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@
     dest: /etc/crio/crio.conf
     line: '"docker.io"'
     insertafter: 'registries = \['
-    regexp: 'docker\.io'
+    regexp: 'registries = \[ "docker\.io" \]'
     state: present
```
Comment (cevich) [inline contrib/test/integration/build/cri-o.yml]: mmmmm, no, I don't think that's right.  Probably also need to remove the ``insertafter: 'registries = \['`` argument.

Comment (cevich) [reply contrib/test/integration/build/cri-o.yml]: ...reading docs...

Comment (cevich) [reply contrib/test/integration/build/cri-o.yml]: yeah, ``lineinfile`` is really messy to use in this case.   If the format of the file changes, breakage is likely, and keeping the regex in-sync is a PITA.  If there's no regex-match, ``insertafter`` will be taken to mean EOF...not what you want. 
 Better just do:

```
- name: blah
  copy:
    dest: /etc/crio/crio.conf
    content: |
         # File added by ansible from build/cri-o.yml
         ....file contents.....
```

File: contrib/test/integration/build/cri-o.yml [hunk 3]
```diff
@@ -64,7 +64,9 @@
 - name: add docker.io default registry
   lineinfile:
     dest: /etc/crio/crio.conf
-    line: '"docker.io"'
+    line: |
+          # Added by Ansible from build/cri-o.yml
+          'registries = [ "docker.io" ]'
```
Comment (cevich) [inline contrib/test/integration/build/cri-o.yml]: I don't think you want the single-quotes here, they will be written into the file.

**Comment by cevich** [comment]
Changes LGTM.  Only thing I'm concerned about is the formatting of the file...``lineinfile`` is basically ``sed``, it's entirely ignorant of the file's syntax.  It looks like ``crio.conf`` is ini-like.  Maybe the ``inifile`` Ansible task could be used, that _is_ respectful of the syntax...unless it's non-standard, then it'll break horribly :confounded:    Trying that out...

**Comment by cevich** [comment]
...findings:  In a fedora container, the default ``/etc/crio.conf`` file is ini-like, but it doesn't do the line-folding in an ansible-friendly way:

```ini
registries = [
]
```

So this:

```yaml
    - ini_file:
        path: /etc/crio/crio.conf
        section: crio.image
        option: registries
        value: '[ "docker.io" ]'

```

results in a ``crio.conf``:

```ini
registries = [ "docker.io" ]
]
```

Which is broken :cry:

Unf. the "standard" for ini-files is...there is no universal standard.  I poked around trying to get Ansible's ``ini_file`` module working correctly, but failed :disappointed:   The only way to get it to work is, if those json-like lists were all jammed into single-lines :confounded:

Modifying CRI-O to support Ansible seems like a horse/cart mixup.  Oh well.  Nobody can say I didn't make an effort :grin:

**Comment by rhatdan** [comment]
@mrunalp @runcom @umohnani8 @wking PTAL
I think this is ready to merge.

**Comment by rhatdan** [comment]
@mrunalp @runcom @umohnani8 @wking PTAL
I think this is ready to merge.

**Comment by rhatdan** [comment]
Merging.

**Pull request merged**